exercise-1.js

```diff
@@ -444,74 +711,148 @@
+                    onlyNumbers.push(3);                    
+                }
+            }
+            return onlyNumbers;
         }
+        let justNumbers = maxNumber(numbersMixed); // NEED TO FINISH
+        console.log(Math.max(...justNumbers));
 
         //After the numbers array has been cleaned up to only have numbers in it, Write a function that sorts the modified numbers array.  Allow the function to sort the array in descending order as well.
 
         function sortNums(numbers,desc=false) {
-            //your code...
+            numbers.sort(function(a, b){return a-b});
+            numbers.sort(function(a, b){return b-a});
         };
+        sortNums(justNumbers);
 
 
 
         /************************************************************* */
         //Research a new feature of ES6+ and create an example of it's use case here.  Be sure to write comments explaining what the feature is and why it is useful.
-
-
             //your code...
-
-
-
+            //ES7
+            // this is the object.entries method. It takes in objects as an argument (Object.entries(CARS)) and truns it into an array of arrays. 
+            // this is useful because it allows you to be able to use the map function with it
+            // the map function takes the entries and maps it for an object.
+            let cars = {
+                honda: 5,
+                mazda: 6,
+                chevy: 7,
+                ford: 9 //additionally, you can add a comma here for the last element. But you dont have too.
+            }
+            for (let [key, val] of Object.entries(cars)) {
+                console.log(key, val);
+            }
         /************************************************************* */
         //Add an example of the 5 primary JavaScript data types to the given mapObj.  The key is an example data type, and the value is the name of the data type.  An object data type has already been set as the 1st key / val pair.
 
         const mapObj = new Map();
-        mapObj.set({company : "TEKsystems"},"object");
-
-        console.log(mapObj.has({company : "TEKsystems"}));
-
+        mapObj.set('company', "TEKsystems");
+        mapObj.set('number', 5);
+        mapObj.set('string', "this is a string");
+        mapObj.set('boolean', true);
+        mapObj.set("array", [1,2,3,4,"hello"]);
+
+        console.log(mapObj.has("company"));
+        console.log(mapObj.has("number"));
         //The above console.log() statmeent returns false.  Write another console.log() statement explaining why this line of code prints false.  Refactor the code `mapObj.set()`, so the code : `mapObj.has() returns true.  The goal is to successfully check and see if {company : "TEKsystems"} exists in the mapObj.
-
+        
         //your code...
-
+        console.log("the key and value are automatically set with the set method based on there location. It converts the keys to strings and therefore you can search by string value");
```

Comment:
The issue was actually that an object described as `{company : "TEKsystems"}` was set as the key of one of the mapped values.  Later, the search is performed using a new object who's key and value are identical to the one in the map, but they are two different objects (they have two different IDs in the system).  The 'has' function looks for an object in the map that has the exact object used in the argument.  The "fix" is to set a variable the an instance of that object, replace the object in the mapped item with the variable, then use that variable in the 'has' call.